core/src/main/java/org/radargun/stages/test/legacy/LegacyTestStage.java

```diff
@@ -96,10 +101,11 @@ public DistStageAck executeOnSlave() {
       try {
          long startNanos = TimeService.nanoTime();
          log.info("Starting test " + testName);
-         List<LegacyStressor> stressors = execute();
+         stressorsManager = setUpAndStartStressors();
+         waitForStressorsToFinish(stressorsManager);
```

Comment:
These two methods are used separately in BackgroundRESTOperationsStartStage and BackgroundRESTOperationsStopStage . I cannot do what you suggest because in the "stop" stage I need to obtain the stressor manager from the "start" stage. So I'm passing it as a parameter.